pandas/core/arrays/datetimes.py

```diff
@@ -1563,6 +1648,24 @@ def sequence_to_dt64ns(data, dtype=None, copy=False,
     # if dtype has an embedded tz, capture it
     tz = validate_tz_from_dtype(dtype, tz)
 
+    if isinstance(data, (ABCSeries, ABCIndexClass)):
+        data = data._data
+
+    if isinstance(data, DatetimeArrayMixin):
+        # series / index have been unboxed. If we're here, we just
+        # need to validate against user-provided parameters and exit early.
+        if tz and data.tz:
+            if not timezones.tz_compare(tz, data.tz):
+                msg = (
+                    "Timezone of the array and 'dtype' do not match. "
+                    "'{}' != '{}'"
+                )
+                raise TypeError(msg.format(tz, data.tz))
+        tz = data.tz
+        tz = validate_tz_from_dtype(dtype, tz)
+
+        return data, tz, None
```

Comment:
this changes the signature to return DTA instead of ndarray.  Any particular reason?

Reply:
Thanks.  If you get a chance to look at the new PR^2 to rebase again, it'd be good for motivation to see that we've accomplished some diff-reduction.

Reply:
I had some trouble with having this return an ndarray.

Reply:
Can you elaborate on this a bit?  Having this function return either a ndarray or a DTA is pretty janky; a big part of why this function was implemented was to make this strictly lower-level than the constructors

Reply:
Hmm may have been from not returning `data.freq`. I'm running the full test suite again.